zkvm/docs/zkvm-spec.md

```diff
@@ -1390,49 +1408,91 @@ Fails if:
 
 #### input
 
-_input_ **input** → _contract_
+_prevoutput_ **input** → _contract_
 
-1. Pops a [data](#data-type) `input` representing the [input structure](#input-structure) from the stack.
-2. Constructs a [contract](#contract-type) based on the `input` data and pushes it to the stack.
+1. Pops a [data](#data-type) `prevoutput` representing the [unspent output structure](#output-structure) from the stack.
+2. Constructs a [contract](#contract-type) based on the `prevoutput` data and pushes it to the stack.
 3. For each decoded [value](#value-type), quantity variable is allocated first, flavor second.
 4. Adds [input entry](#input-entry) to the [transaction log](#transaction-log).
-5. Sets the [VM uniqueness flag](#vm-state) to `true`.
+5. Sets the [VM’s last anchor](#vm-state) to the [contract ID](#contract-id) of the claimed 
+[UTXO](#utxo).
 
-Fails if the `input` is not a [data type](#data-type) with exact encoding of an [input structure](#input-structure).
+Fails if the `prevoutput` is not a [data type](#data-type) with exact encoding of an [output structure](#output-structure).
 
 #### output
 
 _items... predicate_ **output:_k_** → ø
 
 1. Pops [`predicate`](#predicate) from the stack.
 2. Pops `k` items from the stack.
-3. Adds an [output entry](#output-entry) to the [transaction log](#transaction-log).
+3. Use the [VM’s last anchor](#vm-state) as an anchor for the [output structure](#output-structure).
+4. Update the [VM’s last anchor](#vm-state) with a ratcheted [contract ID](#contract-id):
+    ```
+    T = Transcript("ZkVM.ratchet-anchor")
+    T.commit("output", output_id)
```

Comment:
Here, what is `output_id`? Is this the VM's last anchor?

Reply:
rewrote this part here: 
https://github.com/interstellar/slingshot/pull/193/commits/6b995d569476e1bf2b41f03fd3222482f80403bf